File: hash/jewels-and-stones.1.js
var numJewelsInStones = function(jewels, stones) {
  let count = 0; 
  let hashTable = {};
  
  for(let index = 0; index < stones.length; index++){
      if (hashTable[stones[index]] == undefined){
          hashTable[stones[index]] = 1;
         } else{
             hashTable[stones[index]] = hashTable[stones[index]] + 1;
         }
  }
  
  for(let index = 0; index < jewels.length; index++){
      if(hashTable[jewels[index]] !== undefined){
          count += hashTable[jewels[index]];
         } else{
             count += 0;
         }
  }
  
  return count;
};
numJewelsInStones("aA","aAAbbbb")

File: hash/jewels-and-stones.1.java
public class JewelsStones {
    public static void main(String[] args) {
     int com =   numJewelsInStones("aA","aAAbbbb");
        System.out.println(com);
    }

    public static int numJewelsInStones(String jewels, String stones) {
        int number =  0;
        int jewelLength = jewels.length();
        int stoneLength = stones.length();
        for(int i=0;i<jewelLength;i++){
            for(int j=0;j<stoneLength;j++){
                if(jewels.charAt(i) == stones.charAt(j)){
                    number++;
                }
            }
        }
        return number;
    }
}
